solr/core/src/java/org/apache/solr/handler/RequestHandlerBase.java

```diff
@@ -208,14 +208,17 @@ public void handleRequest(SolrQueryRequest req, SolrQueryResponse rsp) {
         }
       }
     } catch (Exception e) {
+      boolean hasTragicException = false;
```

Comment:
Here is what I was thinking of after this statement. What do you think?

        if (isTragic) {
          if (e instanceof SolrException) {
            assert ((SolrException) e).code() == 500; //Tragic exceptions should always throw a server error
          } else {
            //wrap it in a solr exception
            e = new SolrException(SolrException.ErrorCode.SERVER_ERROR, e);
          }
        }

Reply:
Looking at LeaderTragicEventTest.java maybe we should check if the code is a 500 or a 404 also?

Reply:
@vthacker 
Changed the code as you suggested, and like you variable name: isTragic :)
I checked the code is one of 500 or 404 in LeaderTragicEventTest.java for update or commit:
```java
   assertThat(se.code(), anyOf(is(500), is(404)));
```